controllers/openstackmachine_controller.go

```diff
@@ -336,9 +337,14 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, logger
 		openStackMachine.Status.Ready = true
 	case infrav1.InstanceStateBuilding:
 		logger.Info("Machine instance is BUILDING", "instance-id", instance.ID)
-	default:
+	case infrav1.InstanceStateError:
+		// Error is unexpected, thus we report error and never retry
 		handleUpdateMachineError(logger, openStackMachine, errors.Errorf("OpenStack instance state %q is unexpected", instance.State))
 		return ctrl.Result{}, nil
+	default:
+		// The other state is normal (for example, migrating) but we don't want to proceed until it's ACTIVE
+		// due to potential conflict or unexpected actions
+		return ctrl.Result{RequeueAfter: waitForInstanceBecomeActiveToReconcile}, nil
 	}
```

Comment:
to be honst, follow this code, we only do floating ip association which is safe whenever the VM is in which state (the floating ip operation not on VM , but on port) .. so I think it's ok to accept those status (so that means no requeue)

but to be safe, seems only allow ACTIVE is also reasonable (eventually the VM will go ACTIVE ?)

Reply:
Ideally we need to care all statuses: https://docs.openstack.org/api-ref/compute/?expanded=show-server-details-detail,list-servers-detail#list-servers
I found there is no `BUILDING` status.

Reply:
In addtion to `ERROR`, `DELETED`, `UNKNOWN` statuses never become `ACTIVE` again.
I think other statuses could become `ACTIVE` by automatically(`BUILD`) or manually.

Reply:
@hidekazuna BUILDING is correct, 
https://github.com/openstack/nova/blob/master/nova/compute/vm_states.py#L37

Reply:
Not correct. status is determined by `vm_state` and `task_state`.
function: https://github.com/openstack/nova/blob/master/nova/api/openstack/common.py#L115-L124
dictionary: https://github.com/openstack/nova/blob/master/nova/api/openstack/common.py#L45-L112

This means if vm_state is BUILDING, status is always BUILD.

Reply:
you are right, the status is BUILD 
```

nova --debug show 1d36ba54-4dfa-432a-b773-dda237770b28

"status": "BUILD",
...
"OS-EXT-STS:vm_state": "building",
```